internal/processing/status/status.go

```diff
@@ -0,0 +1,45 @@
+package status
+
+import (
+	"github.com/kyma-project/api-gateway/apis/gateway/versions"
+	"github.com/kyma-project/api-gateway/internal/validation"
+)
+
+type ReconciliationStatusVisitor interface {
```

Comment:
To avoid confusion I think we should not name it Visitor, since we don't have a "typical" visitor implementation.
```suggestion
type ReconciliationStatus interface {
```